h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -782,21 +782,25 @@ public AutoBuffer put(Freezable f) {
   }
 
   public <T extends Freezable> T get() {
-    int id = getInt();
-    if( id == TypeMap.NULL ) return null;
-    if( _is!=null ) {
-      id = remapFrozenId(id);
-    }
-    return (T)TypeMap.newFreezable(id).read(this);
+    return getFreezable(null);
   }
+
   public <T extends Freezable> T get(Class<T> tc) {
+    if (tc == null)
+      throw new IllegalArgumentException("Class argument cannot be null");
```

Comment:
get and get(Class<T>) both delegate to getFreezable

but if the caller chooses to use the version with the class argument, the argument shouldn't be null

Does it make sense?

Reply:
yes, it is ensure the caller is not accidentally passing null instead of the class - the only purpose of the class is to be able to check the returned instance is of the expected type

if I start accidentally passing null - I won't trigger the check